test/libsolidity/TestHook.h

```diff
@@ -0,0 +1,52 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "Builtin.h"
+#include <memory>
+#include <vector>
+
+namespace solidity::frontend::test
+{
+
+class TestFunctionCall;
+
+class TestHook
+{
+public:
+	TestHook() = default;
+	virtual ~TestHook() = default;
+
+	virtual void beginTestCase() {}
+	virtual void beforeFunctionCall([[maybe_unused]] TestFunctionCall const& _call) {}
```

Comment:
```suggestion
	virtual void beforeFunctionCall(TestFunctionCall const&) {}
```
Isn't this better readable?

Reply:
I had it like this this first. But then I thought that the parameter names could communicate more meaning of the actual function, if they are present. In this particular case the parameter name can probably be removed. Maybe I will just put the parameter names as comments.